core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/GraphicElementFactory.java

```diff
@@ -187,15 +203,38 @@ public SettingBundle getFavoriteLookSettings() {
    */
   public final void setLook(String lookName) {
     // get the customer lookSettings
-    String selectedLook = lookSettings.getString(lookName, null);
-    if (selectedLook == null) {
-      // ce look n'existe plus, look par defaut
-      selectedLook = defaultLook;
+    String look = lookSettings.getString(lookName, null);
+    if (!isLookAllowed(lookName)) {
+      // this look no more exists, set an existing look
+      lookName = getAvailableLooksForUser().get(0);
+      look = lookSettings.getString(lookName, null);
     }
+    this.favoriteLookSettings = ResourceLocator.getSettingBundle(look);
+    this.currentLookName = lookName;
+  }
 
-    this.favoriteLookSettings = ResourceLocator.getSettingBundle(selectedLook);
+  private static String getExistingLook() {
+    Set<String> lookNames = lookSettings.keySet();
+    for (String aLookName : lookNames) {
+      if (isLookExist(aLookName)) {
+        return aLookName;
+      }
+    }
+    return defaultLookName;
+  }
 
-    currentLookName = lookName;
+  private static boolean isLookExist(String lookName) {
+    try {
+      String selectedLook = lookSettings.getString(lookName);
+      SettingBundle look = ResourceLocator.getSettingBundle(selectedLook);
+      return look.exists();
+    } catch (MissingResourceException e) {
```

Comment:
When an exception is caught, it is expected one of the two behaviors:
- log the exception (in this case: `SilverLogger.error("Look named '"+lookName+" not found", e)`, see the exception e logged at the end of the error method)
- rethrow the exception or throw another exception

When this rules don't need to be applied, then we add an exception to this rule for the incriminated code in our SonarQube.

For your case, in order to determine the better way to handle that, when the method `isLookExist` is called with a look name as argument, should the name exist in the look settings?